helmclient.go

```diff
@@ -425,6 +427,69 @@ func (c *Client) InstallFromTarball(path, ns string, options ...helmclient.Insta
 	return nil
 }
 
+func (c *Client) ListReleaseContents(ctx context.Context) ([]*ReleaseContent, error) {
+	var releases []*hapirelease.Release
+	{
+		o := func() error {
+			t, err := c.newTunnel()
+			if IsTillerNotFound(err) {
+				return backoff.Permanent(microerror.Mask(err))
+			} else if err != nil {
+				return microerror.Mask(err)
+			}
+			defer c.closeTunnel(t)
+
+			moreReleases := true
+			next := ""
+			for moreReleases {
+				// Note: We explicitly ask for all release statuses,
+				// otherwise Helm will only return successfully deployed releases.
+				resp, err := c.newHelmClientFromTunnel(t).ListReleases(
+					helmclient.ReleaseListStatuses(helmStatuses),
+					helmclient.ReleaseListOffset(next),
+				)
+				if err != nil {
+					return microerror.Mask(err)
+				}
+
+				releases = append(releases, resp.GetReleases()...)
+
+				if resp.GetNext() == "" {
+					moreReleases = false
+				} else {
+					next = resp.GetNext()
+				}
```

Comment:
Please just replace with `for {`. There is no need to maintain another variable.

```go
next = resp.GetNext()
if next == "" {
	break
}
```

Reply:
Done